surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -48,15 +54,28 @@ private boolean isUsageSet() {
         for (final String strategy : strategies) {
 
             final TestExecutionPlanner plannerForStrategy = testExecutionPlannerLoader.getPlannerForStrategy(strategy);
-            final List<? extends Class<?>> tests = plannerForStrategy.getTests().stream().map(testClass -> {
-                try {
-                    return Class.forName(testClass);
-                } catch (ClassNotFoundException e) {
-                    throw new IllegalStateException("Failed while obtaining strategy for " + strategy, e);
-                }
-            }).collect(Collectors.toList());
+            final List<? extends Class<?>> tests = plannerForStrategy.getTests()
+                .stream().filter(this::validateClassPathAndTestClass)
+                .map(testClass -> {
+                    try {
+                        return Class.forName(testClass);
+                    } catch (ClassNotFoundException e) {
+                        throw new IllegalStateException("Failed while obtaining strategy for " + strategy, e);
+                    }
+                }).collect(Collectors.toList());
             orderedTests.addAll(tests);
         }
         return orderedTests;
     }
+
+    private boolean validateClassPathAndTestClass(String testClass) {
+        Class<?> aClass;
+        try {
+            aClass = testClassLoader.loadClass(testClass);
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+
+        return jUnit4TestChecker.accept(aClass);
```

Comment:
Otherwise we have to support every possible framework surefire does. And I believe this is not our responsibility as we simply should delegate it to the underlying "real test runner". We are just "test order manipulator" here.